staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go

```diff
@@ -234,48 +261,36 @@ func CreateNewCustomResourceDefinition(crd *apiextensionsv1beta1.CustomResourceD
 	// For this test, we'll actually cycle, "list/watch/create/delete" until we get an RV from list that observes the create and not an error.
 	// This way all the tests that are checking for watches don't have to worry about RV too old problems because crazy things *could* happen
 	// before like the created RV could be too old to watch.
-	var primingErr error
-	wait.PollImmediate(500*time.Millisecond, 30*time.Second, func() (bool, error) {
-		primingErr = checkForWatchCachePrimed(crd, dynamicClientSet)
-		if primingErr == nil {
-			return true, nil
+	for _, v := range getAllServingCRDVersions(crd) {
+		version := v
+		err := wait.PollImmediate(500*time.Millisecond, 30*time.Second, func() (bool, error) {
+			return checkForWatchCachePrimed(crd, dynamicClientSet, version)
+		})
+		if err != nil {
+			return nil, err
 		}
-		return false, nil
-	})
-	if primingErr != nil {
-		return nil, primingErr
 	}
-
 	return crd, nil
 }
 
-func checkForWatchCachePrimed(crd *apiextensionsv1beta1.CustomResourceDefinition, dynamicClientSet dynamic.Interface) error {
+// checkForWatchCachePrimed returns true if the watch is primed for an specified version of CRD watch
+func checkForWatchCachePrimed(crd *apiextensionsv1beta1.CustomResourceDefinition, dynamicClientSet dynamic.Interface, version string) (bool, error) {
```

Comment:
isWatchCachePrimed. It's odd if a command like "Check" returns a bool. The func should be named like a predicate.